### PR TITLE
Add how to supply ext args with params

### PR DESCRIPTION
**Pull request by mahesh-panchal**
Add description to website how to use params to supply nextflow config in modules https://github.com/nf-core/modules/pull/6788

@netlify /docs/tutorials/tests_and_test_data/nf-test_writing_tests

### CODE REVIEW

**Review by jfy133 — APPROVED** [review]

File: sites/docs/src/content/docs/tutorials/tests_and_test_data/nf-test_writing_tests.md [hunk 1]
```diff
@@ -12,21 +12,49 @@ weight: 10
...
 
-Then add the path to the `main.nf.test` file.
+```groovy title="nextflow.config"
+process {
+  withName: 'MODULE' {
+    ext.args = params.module_args
+  }
+}
+```
+
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/tests_and_test_data/nf-test_writing_tests.md:30]:
```suggestion

You do not need to modify the contents of this file any further. 

```

**Comment by jfy133** [comment]
And should add mention of this in teh modules specs

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 2]
```diff
@@ -750,6 +750,37 @@ Input data SHOULD be referenced with the `modules_testdata_base_path` parameter:
 file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true)
 ```
 
+### Configuration of ext.args in tests
+
+Module nf-tests SHOULD use a single `nextflow.config` to supply `ext.args` to a module, by taking the settings from the
+params block defined in the `when` block of a test.
```
Comment (JoseEspinosa) [inline sites/docs/src/content/docs/guidelines/components/modules.md:756]: To me the sentence was difficult to read that is why I tried to change it a little
```suggestion
Module nf-tests SHOULD use a single `nextflow.config` to supply `ext.args` to a module. They can be defined in the `when` block of a test under the `params` scope.
```

**Review by JoseEspinosa — APPROVED** [review]
LGTM!

I just add a small suggestion

**Review by jfy133 — COMMENTED** [review]
I forgot to finish a review, sorry @mahesh-panchal

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 3]
```diff
@@ -750,6 +750,37 @@ Input data SHOULD be referenced with the `modules_testdata_base_path` parameter:
...
+    input[0] = [
+      [ id:'test1', single_end:false ], // meta map
+      file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
+    ]
+    """
+  }
+}
+```
+
+```groovy title="nextflow.config"
```
[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: I added something, although I don't think it's really necessary.

The syntax highlighting looked OK to me. Can you describe what's off?

[the remark this answers is not in the document]
Comment (jfy133) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: I haven't checked the deploy preview admittedly, but in the github reviewing context, the normal text that should be white is blue like it thinks it's code:

image

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: Ah, OK. I think I found it around line 184. I'll see if it fixes it.

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: There's something I don't understand on Line 193 now that changes formatting.

[the remark this answers is not in the document]
Comment (mahesh-panchal) [reply sites/docs/src/content/docs/guidelines/components/modules.md]: @mashehu Any idea why the highlighting goes funny on Line 196?

**Review by edmundmiller — APPROVED** [review]
Awesome idea. I think it'll help cut down on the "where did that come from?" feeling of the tests.

More importantly do we have an issue to create linting around this?

**Comment by mashehu** [comment]
> More importantly do we have an issue to create linting around this?

what exactly should we lint for?

**Comment by mahesh-panchal** [comment]
> > More importantly do we have an issue to create linting around this?
> 
> what exactly should we lint for?

A single `nextflow.config` file in tests, and that it contains the string `ext.args = params.`, and not `ext.args = [{'"]`

**Comment by jfy133** [comment]
@mahesh-panchal you can merge in now, unless you want to wait for the linting test

**Pull request merged**